File: LaboRTI2023_sources/Data/AchatArticleException.cpp
#include "AchatArticleException.h"

const int AchatArticleException::INSUFFICIENT_STOCK = 3;

AchatArticleException::AchatArticleException() : Exception() { }

AchatArticleException::AchatArticleException(string msg, int code) : Exception(msg)
{
    this->setCode(code);
}

void AchatArticleException::setCode(int code)
{
    if(code == INSUFFICIENT_STOCK)
    {
        this->code = code;
    }
}

int AchatArticleException::getCode() { return this->code; }

File: LeMaraicher_ClientAchat/src/Modele/Exceptions/AchatArticleException.java
package Modele.Exceptions;

public class AchatArticleException extends Exception {

    public static final int INSUFFICIENT_STOCK = 3;
    private int code;

    public AchatArticleException() {
        super();
    }

    public AchatArticleException(String msg, int code) {
        super(msg);
        this.code = code;
    }

    public int getCode() {
        return code;
    }
}
